playwright-tests/browserstack.config.ts

```diff
@@ -0,0 +1,66 @@
+import 'dotenv/config';
+import { exec } from 'child_process';
+import BrowserStackLocal from 'browserstack-local';
+
+interface BrowserDetails {
+	browserName: string;
+	browserVersion?: string;
+	osName: string;
+	osVersion?: string;
+}
+
+interface BrowserCapabilities {
+	browser: string;
+	browser_version?: string;
+	os: string;
+	os_version: string;
+	name: string;
+	build: string;
+	'browserstack.username': string;
+	'browserstack.accessKey': string;
+	'browserstack.local': boolean;
+	'client.playwrightVersion': string;
+}
+
+const clientPlaywrightVersion = exec('npx playwright --version')
+	.toString()
+	.trim()
+	.split(' ')[1] as string;
+
+const caps: BrowserCapabilities = {
+	browser: 'chrome',
+	os: 'osx',
+	os_version: 'catalina',
+	name: 'My first playwright test',
+	build: 'playwright-build',
+	'browserstack.username': process.env.BROWSERSTACK_USERNAME ?? '',
+	'browserstack.accessKey': process.env.BROWSERSTACK_ACCESS_KEY ?? '',
+	'browserstack.local': false,
+	'client.playwrightVersion': clientPlaywrightVersion,
+};
```

Comment:
I wonder if it's worth being more explicit here about what properties are set, and which are configurable via calling `getCdpEndpoint`? This would also avoid a little mutation which can sometimes be hard to debug.

Keeping the names the same to avoid levels of indirection:

```suggestion
const browserStackProps: BrowserStackProps = {
	build: 'playwright-build',
	'browserstack.username': process.env.BROWSERSTACK_USERNAME ?? '',
	'browserstack.accessKey': process.env.BROWSERSTACK_ACCESS_KEY ?? '',
	'browserstack.local': false,
	'client.playwrightVersion': clientPlaywrightVersion,
};

type BrowserDetails = {
  name: string;
	browser_version: string;
	os: string;
	osVersion?: string;
}
```

and then just join them in `getCdpEndpoint`:
```ts
const cap = {
  ...browserStackProps,
  ...browserDetails
}
const cdpUrl = `wss://cdp.browserstack.com/playwright?caps=${encodeURIComponent(
  JSON.stringify(caps),
)}`;
```